### ISSUE TITLE
[PC] Vulkan validation error: UNASSIGNED-CoreValidation-Shader-InconsistentSpirv

### ISSUE THREAD

**Issue by moraaar** [o3de/o3de#11366]
**Steps to reproduce**
1. Build editor and launcher on PC on profile
2. Run Asset Processor to build assets
4. Run a default level with the launcher using options "-rhi=vulkan -rhi-device-validation=enable"
5. Check log

**Expected behavior**
No vulkan errors

**Actual behavior**
````
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x20f49cc7af8, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: Structure id 3 decorated as BufferBlock for variable in Uniform storage class must follow relaxed storage buffer layout rules: member 0 contains an array with stride 180 not satisfying alignment to 16
  %type_RWStructuredBuffer_DDGIVolumeDescGPU = OpTypeStruct %_runtimearr_DDGIVolumeDescGPU
````

**Found in Branch**
Development f1d46fc8f7626f5bcac919013d3ebde3d8551e40

**Desktop/Device (please complete the following information):**
 - Device: [PC]
 - OS: [Windows]
 - Version [10]
 - CPU [Ryzen Threadripper 3970X]
 - GPU [NVidia RTX 2080Ti]
 - Memory [48GB]

**Comment by moudgils** [comment]
@dmcdiarmid-ly I think this can be closed as it is fixed. Can you confirm?

**Comment by mythrz** [comment]
Edit2: machine 1, 1060 gtx and amd ryzen 7 2700x.

---

To avoid creating a completely new issue, since the reproduction is exactly the same, I will post my output. I only noticed these errors with commit b19a4172dc452342640c01e43afe8e8c4e7ef8ef, development-branch, with debug-build

running with -rhi-device-validation=enable, but not having this line show the same errors! So maybe this is on by default, or the errors are appearing for another reason.

These are visible before:
```bash
(EMotionFX) - EMotion Studio initialized...
> r_shadowCascadeExtrusionAmount : -1
> r_enablePerMeshShaderOptionFlags : false
> r_meshInstancingEnabled : false
> r_meshInstancingEnabledForTransparentObjects : false
> r_meshInstancingBucketSortScatterBatchSize : 512
```
and
```bash
<08:20:31> Executing console batch file (try game,config,root): "user.cfg" found in game/config/ ...
<08:20:31> (RHI) - ****************************************************************
<08:20:31> (RHI) -                     Registering vulkan RHI
<08:20:31> (RHI) - ****************************************************************
```
but can be a coincidence. Other people would have to report similar events. (these warnings and errors repeat several times, after the above two headers start).


```bash
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-VkShaderModuleCreateInfo-pCode-01091 ] Object 0: handle = 0x562eff090070, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xa7bb8db6 | vkCreateShaderModule(): The SPIR-V Capability (RayQueryKHR) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the capabilities listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkShaderModuleCreateInfo-pCode-01091)
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-VkShaderModuleCreateInfo-pCode-04147 ] Object 0: handle = 0x562eff090070, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x3d492883 | vkCreateShaderModule(): The SPIR-V Extension (SPV_KHR_ray_query) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the SPIR-V extensions listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkShaderModuleCreateInfo-pCode-04147)
[Warning] (vkDebugMessage) - [WARNING][Performance] Validation Performance Warning: [ UNASSIGNED-CoreValidation-Shader-OutputNotConsumed ] Object 0: handle = 0xcc28470000000642, type = VK_OBJECT_TYPE_SHADER_MODULE; | MessageID = 0x609a13b | vertex shader writes to output location 0.0 which is not consumed by fragment shader
[Warning] (vkDebugMessage) - [WARNING][Performance] Validation Performance Warning: [ UNASSIGNED-CoreValidation-Shader-OutputNotConsumed ] Object 0: handle = 0xd0258d0000000648, type = VK_OBJECT_TYPE_SHADER_MODULE; | MessageID = 0x609a13b | vertex shader writes to output location 0.0 which is not consumed by fragment shader
[Warning] XML reader: Can't open file (/home/d/O3DE/Projects/SystemPrototypes/user/Sandbox/Macros.xml)
Document created
Adding '*$tmp*' to ignore masks for changed files.
[Warning] (Application) - 'IsPrefabSystemEnabled' is deprecated, the editor only supports prefabs for level editing.
```

This warning appears in the middle of the above errors and warnings too, but only once.

```bash
<08:20:40> (Asset Browser DEBUG) - 37607/37607 data changes applied
<08:20:40> [Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for translationassets/types/ondemandreflectedtypes.names took 664 ms to load 4010690 bytes (    -nan KB/s)
<08:20:40> [Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for shaders/reflections/reflectionscreenspaceraytracingclosesthitprocedural.azshader took 119 ms to load 1068391 bytes (    -nan KB/s)
```

This one occurs later. It appears to be trying to process a model before it is ready to be processed:

```bash
<08:21:12> [Warning] (Model) - ray intersection against a model that is still creating spatial information
<08:21:13> [Warning] (vkDebugMessage) - [WARNING][Performance] Validation Performance Warning: [ UNASSIGNED-CoreValidation-Shader-OutputNotConsumed ] Object 0: handle = 0xd91f700000001b51, type = VK_OBJECT_TYPE_SHADER_MODULE; | MessageID = 0x609a13b | Vertex attribute at location 4 not consumed by vertex shader
<08:21:13> [Warning] (vkDebugMessage) - [WARNING][Performance] Validation Performance Warning: [ UNASSIGNED-CoreValidation-Shader-OutputNotConsumed ] Object 0: handle = 0xf1878f0000001b67, type = VK_OBJECT_TYPE_SHADER_MODULE; | MessageID = 0x609a13b | vertex shader writes to output location 6.0 which is not consumed by fragment shader

<08:21:13> [Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0xd0561b0000000beb, name = m_objectToWorldBuffer0, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0xd0561b0000000beb[m_objectToWorldBuffer0] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
<08:21:13> [Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x5e963f0000000ff4, name = MaterialParameterBuffer_00, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x5e963f0000000ff4[MaterialParameterBuffer_00] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
<08:21:13> [Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x1366f00000001ad5, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x1366f00000001ad5[] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)

<08:24:37> [Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-VkShaderModuleCreateInfo-pCode-01091 ] Object 0: handle = 0x562eff090070, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xa7bb8db6 | vkCreateShaderModule(): The SPIR-V Capability (RayQueryKHR) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the capabilities listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkShaderModuleCreateInfo-pCode-01091)
<08:24:37> [Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-VkShaderModuleCreateInfo-pCode-04147 ] Object 0: handle = 0x562eff090070, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x3d492883 | vkCreateShaderModule(): The SPIR-V Extension (SPV_KHR_ray_query) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the SPIR-V extensions listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkShaderModuleCreateInfo-pCode-04147)

<08:24:38> [Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x6b15d000000cfb0, name = LightList0, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x6b15d000000cfb0[LightList0] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
```

**Comment by mythrz** [comment]
The second machine (laptop with GP108M GeForce MX150 and intel i5 8th gen), testing with the same validation "build/linux/bin/profile/Editor -rhi-device-validation=enable"

Note that without -rhi-device-validation=enable, no error appears on this machine. This is also a clean engine + project with NO custom assets in it, so there is nothing "tainted" in these results.

```bash
(EMotionFX) - EMotion Studio initialized...
> r_shadowCascadeExtrusionAmount : -1
> r_enablePerMeshShaderOptionFlags : false
> r_meshInstancingEnabled : false
> r_meshInstancingEnabledForTransparentObjects : false
> r_meshInstancingBucketSortScatterBatchSize : 512
[Warning] (vkDebugMessage) - [WARNING][Performance] Validation Performance Warning: [ UNASSIGNED-CoreValidation-Shader-OutputNotConsumed ] Object 0: handle = 0x9ab2d400000005fc, type = VK_OBJECT_TYPE_SHADER_MODULE; | MessageID = 0x609a13b | vertex shader writes to output location 0.0 which is not consumed by fragment shader

...

[Warning] XML reader: Can't open file (/home/d/O3DE/Projects/SystemPrototypes/user/Sandbox/Macros.xml)
Document created
Adding '*$tmp*' to ignore masks for changed files.
[Warning] (Application) - 'IsPrefabSystemEnabled' is deprecated, the editor only supports prefabs for level editing.

...

[Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for translationassets/types/ondemandreflectedtypes.names took 251 ms to load 4010690 bytes (  0,0000 KB/s)

[Warning] (FrameGraph) - Invalid State: attachment 'Root.MainPipeline_0.PipelineOutput' was added but never used!
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkAcquireNextImageKHR-swapchain-01802 ] Object 0: handle = 0x980b0000000002e, type = VK_OBJECT_TYPE_SWAPCHAIN_KHR; | MessageID = 0x3e97a888 | vkAcquireNextImageKHR: Application has already previously acquired 2 images from swapchain. Only 2 are available to be acquired using a timeout of UINT64_MAX (given the swapchain has 3, and VkSurfaceCapabilitiesKHR::minImageCount is 2). The Vulkan spec states: If the number of currently acquired images is greater than the difference between the number of images in swapchain and the value of VkSurfaceCapabilitiesKHR::minImageCount as returned by a call to vkGetPhysicalDeviceSurfaceCapabilities2KHR with the surface used to create swapchain, timeout must not be UINT64_MAX (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkAcquireNextImageKHR-swapchain-01802)
[Warning] (Model) - ray intersection against a model that is still creating spatial information
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x6d4b790000000b8d, name = m_objectToWorldBuffer0, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x6d4b790000000b8d[m_objectToWorldBuffer0] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x508259000000106f, name = MaterialTypeBufferIndicesBuffer0, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x508259000000106f[MaterialTypeBufferIndicesBuffer0] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
```

**Comment by mythrz** [comment]
debug instead of profile build shows more info (laptop with GP108M GeForce MX150 and intel i5 8th gen, "build/linux/bin/debug/Editor -rhi-device-validation=enable")

This warning triggers several times for different object handles
```bash
[Warning] (vkDebugMessage) - [WARNING][Performance] Validation Performance Warning: [ UNASSIGNED-CoreValidation-Shader-OutputNotConsumed ] Object 0: handle = 0x9ab2d400000005fc, type = VK_OBJECT_TYPE_SHADER_MODULE; | MessageID = 0x609a13b | vertex shader writes to output location 0.0 which is not consumed by fragment shader
....
[Warning] XML reader: Can't open file (/home/d/O3DE/Projects/SystemPrototypes/user/Sandbox/Macros.xml)

```

```bash
(RHI) - ****************************************************************
(RHI) -                     Registering vulkan RHI
(RHI) - ****************************************************************
> r_shadowCascadeExtrusionAmount : -1
> r_enablePerMeshShaderOptionFlags : false
> r_meshInstancingEnabled : false
> r_meshInstancingEnabledForTransparentObjects : false
> r_meshInstancingBucketSortScatterBatchSize : 512
(Asset Browser DEBUG) - 17472/17472 data changes applied
[Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for translationassets/classes/editortransformbus.names took 207 ms to load 239 bytes (    -nan KB/s)
[Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for translationassets/classes/lightingpreset.names took 208 ms to load 16421 bytes (    -nan KB/s)
[Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for translationassets/ebus/senders/uilayoutgridbus.names took 208 ms to load 12061 bytes (    -nan KB/s)
[Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for translationassets/types/ondemandreflectedtypes.names took 761 ms to load 4010690 bytes (    -nan KB/s)
```

It appears the error is not related to any of my custom models or materials, since this laptop has none of those. The difference was on debug vs profile builds! It occurs when the editor is opened, and every time the game starts

```bash
Level loading time: 0,04 seconds
-----------------------------------------------------------
[Warning] (Model) - ray intersection against a model that is still creating spatial information
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x7cab520000000b86, name = m_objectToWorldBuffer0, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x7cab520000000b86[m_objectToWorldBuffer0] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x55e9210000001067, name = MaterialTypeBufferIndicesBuffer0, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x55e9210000001067[MaterialTypeBufferIndicesBuffer0] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
Entered game mode
(Spawnables) - Entities from new root spawnable 'Root.spawnable' are ready (generation: 0).
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x4ba57b000000173c, name = 0, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x4ba57b000000173c[0] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
Exited game mode
(Asset Browser DEBUG) - 1/1 data changes applied
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0xff333b0000006f6b, name = 0, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0xff333b0000006f6b[0] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
[Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-vkDestroyBuffer-buffer-00922 ] Object 0: handle = 0x5835d30000001601, name = DirectionalLightShadowBuffer(MainPipeline_0)190, type = VK_OBJECT_TYPE_BUFFER; | MessageID = 0xe4549c11 | Cannot free VkBuffer 0x5835d30000001601[DirectionalLightShadowBuffer(MainPipeline_0)190] that is in use by a command buffer. The Vulkan spec states: All submitted commands that refer to buffer, either directly or via a VkBufferView, must have completed execution (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-vkDestroyBuffer-buffer-00922)
```

**Comment by mythrz** [comment]
Following the trails...

```bash
<17:06:52> [Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for shaders/reflections/reflectionscreenspaceraytracingclosesthitprocedural.azshader took 119 ms to load 1068391 bytes (    -nan KB/s)
<17:06:52> [Error] (vkDebugMessage) - [ERROR][Validation] Validation Error: [ VUID-VkShaderModuleCreateInfo-pCode-01091 ] Object 0: handle = 0x5594177e0070, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xa7bb8db6 | vkCreateShaderModule(): The SPIR-V Capability (RayQueryKHR) was declared, but none of the requirements were met to use it. The Vulkan spec states: If pCode declares any of the capabilities listed in the SPIR-V Environment appendix, one of the corresponding requirements must be satisfied (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkShaderModuleCreateInfo-pCode-01091)
```

Could it be these shaders?

https://github.com/o3de/o3de/blob/development/Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceRayTracingClosestHitProcedural.shader

"Vulkan driver reports version 1.4 (which includes the specification for ray tracing), but Pascal GPUs physical hardware lacks the necessary execution units to run RayQueryKHR instructions"
https://forums.developer.nvidia.com/t/rayquerysupportedfeatures-rayquery-not-support-gtx-1060/280665

Since the engine is including multiple shaders and those shaders are calling APIs that may or may not have the required requirements, the errors trigger. I am guessing there is no filter during build of the engine/editor to tell it which shader-variants it should include.

@mprettner-huawei could you please confirm if I am on the right track? And what would you advise doing in these cases?

vulkaninfo --summary

```bash
vulkaninfo --summary
==========
VULKANINFO
==========

Vulkan Instance Version: 1.4.341


Instance Extensions: count = 24
-------------------------------
VK_EXT_acquire_drm_display             : extension revision 1
VK_EXT_acquire_xlib_display            : extension revision 1
VK_EXT_debug_report                    : extension revision 10
VK_EXT_debug_utils                     : extension revision 2
VK_EXT_direct_mode_display             : extension revision 1
VK_EXT_display_surface_counter         : extension revision 1
VK_EXT_surface_maintenance1            : extension revision 1
VK_EXT_swapchain_colorspace            : extension revision 5
VK_KHR_device_group_creation           : extension revision 1
VK_KHR_display                         : extension revision 23
VK_KHR_external_fence_capabilities     : extension revision 1
VK_KHR_external_memory_capabilities    : extension revision 1
VK_KHR_external_semaphore_capabilities : extension revision 1
VK_KHR_get_display_properties2         : extension revision 1
VK_KHR_get_physical_device_properties2 : extension revision 2
VK_KHR_get_surface_capabilities2       : extension revision 1
VK_KHR_portability_enumeration         : extension revision 1
VK_KHR_surface                         : extension revision 25
VK_KHR_surface_protected_capabilities  : extension revision 1
VK_KHR_wayland_surface                 : extension revision 6
VK_KHR_xcb_surface                     : extension revision 6
VK_KHR_xlib_surface                    : extension revision 6
VK_LUNARG_direct_driver_loading        : extension revision 1
VK_NV_display_stereo                   : extension revision 1

Instance Layers: count = 6
--------------------------
VK_LAYER_NV_optimus               NVIDIA Optimus layer         1.4.312  version 1
VK_LAYER_NV_present               NVIDIA GR2608 layer          1.4.312  version 1
VK_LAYER_VALVE_steam_fossilize_32 Steam Pipeline Caching Layer 1.3.207  version 1
VK_LAYER_VALVE_steam_fossilize_64 Steam Pipeline Caching Layer 1.3.207  version 1
VK_LAYER_VALVE_steam_overlay_32   Steam Overlay Layer          1.3.207  version 1
VK_LAYER_VALVE_steam_overlay_64   Steam Overlay Layer          1.3.207  version 1

Devices:
========
GPU0:
        apiVersion         = 1.4.312
        driverVersion      = 580.142.0.0
        vendorID           = 0x10de
        deviceID           = 0x1c03
        deviceType         = PHYSICAL_DEVICE_TYPE_DISCRETE_GPU
        deviceName         = NVIDIA GeForce GTX 1060 6GB
        driverID           = DRIVER_ID_NVIDIA_PROPRIETARY
        driverName         = NVIDIA
        driverInfo         = 580.142
        conformanceVersion = 1.4.1.3
```
